### PR TITLE
KAFKA-1614: Partition log directory and segments info exposed via JMX

### PR DESCRIPTION
**Pull request by everpcpc**
An updated version for https://github.com/apache/kafka/pull/1196

### CODE REVIEW

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -277,6 +278,24 @@ class Log(@volatile var dir: File,
     },
     tags)
 
+  newGauge("LogSegments",
+    new Gauge[util.List[String]] {
+      def value = {
+        val list = logSegments.toSeq.map { seg =>
+          s"baseOffset=${seg.baseOffset}, created=${seg.createdMs}, logSize=${seg.size}, indexSize=${seg.indexSize}"
```
Comment (harshach) [inline core/src/main/scala/kafka/log/Log.scala]: There is a comment on your previous PR about using a different delimiter. Can you address that.

Comment (everpcpc) [reply core/src/main/scala/kafka/log/Log.scala]: Would `|` be OK?

Comment (DarrenBishop) [reply core/src/main/scala/kafka/log/Log.scala]: Perhaps a ';' (semi-colon)?
```suggestion
          s"baseOffset=${seg.baseOffset}; created=${seg.createdMs}; logSize=${seg.size}; indexSize=${seg.indexSize}"
```

**Comment by DarrenBishop** [comment]
What's the process for getting this (and perhaps its predecessor) PRs merged?
What needs to be addressed with the build checks?

File: core/src/main/scala/kafka/log/Log.scala (hunk 1 above)
Comment (everpcpc) [reply core/src/main/scala/kafka/log/Log.scala]: Seems better. Updated.

Comment (everpcpc) [reply core/src/main/scala/kafka/log/Log.scala]: Seems better. Updated.

**Comment by everpcpc** [comment]
The build check error seems not related to the changes.

**Comment by vykimo** [comment]
still not on master branch?

**Comment by m1losz** [comment]
is this gonna get merged to master soon?

**Pull request closed without merging**